src/inmanta/ast/statements/call.py

```diff
@@ -253,6 +253,7 @@ def __init__(self, queue_scheduler, resolver, result: ResultVariable, requires,
         self.result = result
         result.set_provider(self)
         self.requires = requires
+        self.base_requires = dict(requires)
```

Comment:
Perhaps add a comment here stating that `requires` is a dynamic property that is meant to keep track of all execute-time requirements while `base_requires` only contains the direct emit-time requirements.

Reply:
added